src/Loop.php

```diff
@@ -430,3 +455,10 @@ private function __construct()
         // intentionally left blank
     }
 }
+
+// Reset the $level in order to be able to use ::run() inside a shutdown handler
+$f = function()
+{
+    self::$level = 0;
+};
+register_shutdown_function($f->bindTo(null, Loop::class));
```

Comment:
I'm not sure the limited reward justifies the added complexity here, especially since the invocation order of shutdown callbacks can't even be relied upon.